Source/JavaScriptCore/builtins/PromiseConstructor.js

```diff
@@ -368,6 +368,19 @@ function resolve(value)
     return @promiseResolve(this, value);
 }
 
+function withResolvers()
+{
+    "use strict";
+
+    var promiseCapability = @newPromiseCapability(this);
+
+    return {
+        promise: promiseCapability.@promise,
+        resolve: promiseCapability.@resolve,
+        reject: promiseCapability.@reject,
+    };
```

Comment:
I think we should replace all of `@promise` with `promise`, `@resolve` with resolve, `@reject` with reject and making this function as follows. Currently, `newPromiseCapability` is allocating unnecessary object here.

```
function withResolvers()
{
    "use strict";
    return @newPromiseCapability(this);
}
```

Reply:
But keep in mind that `newPromiseCapability`'s function name itself must be `@resolve` etc. to avoid naming them as `"resolve"`.

Reply:
are `@promise`/`@resolve`/`@reject` referenceable outside of builtins?  i tried doing as you suggested (i.e. just `return @newPromiseCapability(this);`) and the resulting object had no properties on it 🤔

or are you suggesting that we rework `newPromiseCapability` itself to return `promise`/`resolve`/`reject` instead (and updating all the callsites to reflect that too)?

Reply:
He's suggesting the latter. To return `promise/resolve/reject` and rework the other callsites.

Reply:
would it be OK if i did that in a followup?  AFAICT looking at it right now it seems like a _lot_ of places to update (as well as other things to change, such as replacing `vm.propertyNames->promisePrivateName()` with something else)

Reply:
to elaborate, id rather this PR just be about adding this new feature and have a separate PR for optimizing it (since AFAICT that optimization involves changing a _lot_ of other things, which probabilistically could have more issues since it's a larger change)

also, is there concern about `newPromiseCapability` no longer matching the spec in the sense that we're using plain properties `promise`/`resolve`/`reject` instead of internal slots `[[Promise]]`/`[[Resolve]]`/`[[Reject]]` (unless im mistaken in thinking that our `@promise`/`@resolve`/`@reject` is equivalent to that)?

Reply:
https://github.com/WebKit/WebKit/pull/15808